spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/config/secretsmanager/SecretsManagerConfigDataLoader.java

```diff
@@ -46,14 +48,23 @@ public SecretsManagerConfigDataLoader(DeferredLogFactory logFactory) {
 	@Nullable
 	public ConfigData load(ConfigDataLoaderContext context, SecretsManagerConfigDataResource resource) {
 		try {
-			SecretsManagerClient sm = context.getBootstrapContext().get(SecretsManagerClient.class);
-			SecretsManagerPropertySource propertySource = resource.getPropertySources()
-					.createPropertySource(resource.getContext(), resource.isOptional(), sm);
-			if (propertySource != null) {
-				return new ConfigData(Collections.singletonList(propertySource));
+			// resource is disabled if secrets manager integration is disabled via
+			// spring.cloud.aws.secretsmanager.enabled=false
+			if (resource.isEnabled()) {
```

Comment:
I would extract everything under if block in new method.

Reply:
I am not a fan of private methods. Not critical, merging as it is.